stdlib/hashtbl.ml

```diff
@@ -352,23 +409,38 @@ module type S =
   sig
     type key
     type 'a t
-    val create: int -> 'a t
+    val create : int -> 'a t
     val clear : 'a t -> unit
     val reset : 'a t -> unit
-    val copy: 'a t -> 'a t
-    val add: 'a t -> key -> 'a -> unit
-    val remove: 'a t -> key -> unit
-    val find: 'a t -> key -> 'a
-    val find_all: 'a t -> key -> 'a list
+    val copy : 'a t -> 'a t
+    val add : 'a t -> key -> 'a -> unit
+    val remove : 'a t -> key -> unit
+    val find : 'a t -> key -> 'a
+    val find_all : 'a t -> key -> 'a list
     val replace : 'a t -> key -> 'a -> unit
     val mem : 'a t -> key -> bool
-    val iter: (key -> 'a -> unit) -> 'a t -> unit
+    val iter : (key -> 'a -> unit) -> 'a t -> unit
     val filter_map_inplace: (key -> 'a -> 'a option) -> 'a t -> unit
-    val fold: (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
-    val length: 'a t -> int
+    val fold : (key -> 'a -> 'b -> 'b) -> 'a t -> 'b -> 'b
+    val length : 'a t -> int
     val stats: 'a t -> statistics
   end
 
+module type FULL =
```

Comment:
The problem is that `Ephemeron` re-uses `Hashtbl.S`, and I'm not sure it is possible to implement the iterator functions on ephemerons (I don't know the module since it's new). Therefore, to avoid copy-pasting, I need to keep the old signature so that `Ephemeron` can refer to it.

Of course the names can change (`FULL` could be `S`, and `S` could be `CORE_S` or whatever).